invoice/src/main/java/org/killbill/billing/invoice/dao/ExistingInvoiceMetadata.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2014-2019 Groupon, Inc
+ * Copyright 2014-2019 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.invoice.dao;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import org.killbill.billing.callcontext.InternalTenantContext;
+import org.killbill.billing.invoice.api.Invoice;
+import org.killbill.billing.invoice.api.InvoiceItem;
+
+public class ExistingInvoiceMetadata {
```

Comment:
Nit: the naming does not really convey what this object does, but i know this is difficult... maybe something like `InvoiceCacheLoader` ?

Reply:
Indeed, it's not really good - but I cannot think of anything else 😖 However, I don't like `InvoiceCacheLoader` because `*CacheLoader` refers to caching loaders already in Kill Bill (and this is unrelated to caches).

I'll merge as-is for now, but let's keep thinking...